File: POO PHP/Aula12/Ave.php
<?php

require_once 'Animal.php';

class Ave extends Animal{

    //Atributo da Classe
    private $corPena;
    
    //Metodo Acessor e Modificador
    public function getCorPena() {
        return $this->corPena;
    }
    public function setCorPena($corPena): void {
        $this->corPena = $corPena;
    }

    //Metodos Sobrepostos da Classe Animal
    public function alimentar() {
        echo "<p>Comendo graos no chao ...</p>";
    }
    public function emitirSom() {
        echo "<p>Emitindo muitos Piu, piu ...</p>";
    }
    public function locomover() {
        echo "<p>Voando ...</p>";
    }

    //Metodo da Classe
    public function fazendoNinho() {
        echo "<p>Construindo ninho na arvore ...</p>";
    }
}

File: POO JAVA/Aula12/src/aula12/Ave.java
package aula12;

public class Ave extends Animal{

    //Atributo da Classe
    private String corPena;
    
    //Metodo Acessor e Modificador
    public String getCorPena() {
        return corPena;
    }
    public void setCorPena(String corPena) {
        this.corPena = corPena;
    }
    
    //Metodo da Classe
    public void fazerNinho(){
        System.out.println("Fazendo ninho na arvore."); 
    }
    
    //Metodos de Sobreposição da Classe Animal
    @Override
    public void locomover() {
        System.out.println("Voando alto ...");
    }
    @Override
    public void alimentar() {
        System.out.println("Comendo graos no chao ...");
    }
    @Override
    public void emitirSom() {
        System.out.println("Piando muito ...");
    }    
}
